pkg/controller/storagecluster/reconcile.go

```diff
@@ -406,6 +407,11 @@ func (r *ReconcileStorageCluster) validateStorageDeviceSets(sc *ocsv1.StorageClu
 				return fmt.Errorf("failed to validate StorageDeviceSet %d: no StorageClass specified for metadataPVCTemplate", i)
 			}
 		}
+		if ds.DeviceType != "" {
+			if !strings.EqualFold(ds.DeviceType, "SSD") || !strings.EqualFold(ds.DeviceType, "HDD") || !strings.EqualFold(ds.DeviceType, "NVMe") {
```

Comment:
Don't just compare strings here. Set them as consts and create an array of valid types. Also, are they actually case-sensitive/mixed case, or can they be all lowercase?

Reply:
they all can be lower case also. mixed case can be for only NVMe. It is just a logical grouping which will added as a metadata for osd.

Reply:
If that's the case then you don't nee dto codify the multiple case types. Just force the string to always be lowercase and compare those values.

Reply:
ok. I'll update the PR

Reply:
You have not finished the requested changes. Please define the strings as consts, preferably with a new `type DeviceType string`.